src/util/patientUtils.js

```diff
@@ -106,4 +108,30 @@ const retrievePatientIds = async ({ subject, organization, practitioner }) => {
   }
 };
 
-module.exports = { getPatientDataCollectionBundle, getPatientDataSearchSetBundle, getPatientData, retrievePatientIds };
+/**
+ * Takes in a Group resource and practitioner from an evaluate-measure query and filters the
+ * patients from the Group to those which reference the Practitioner resource
+ * @param {Object} group A Group Resource
+ * @param {string} practitioner A reference to a FHIR Practitioner. All patients which list the referenced practitioner
+ * as their generalPractitioner will be selected for gaps calculation run on them
+ * @returns array of patient IDs
+ */
+const filterPatientIdsFromGroup = async (group, practitioner) => {
+  const patientPromises = group.member.map(async m => {
+    const query = {
+      id: m.entity.reference.split('/')[1],
+      'generalPractitioner.identifier.value': practitioner
+    };
+    return await findOneResourceWithQuery(query, 'Patient');
+  });
+  const patients = (await Promise.all(patientPromises)).filter(a => a);
```

Comment:
I think this line could use a comment. Took me a bit to figure out that this was filtering out the potentially empty search results. Or, consider changing this map + filter to a reduce

Reply:
[3cc81eb](https://github.com/projecttacoma/deqm-test-server/pull/96/commits/3cc81eb3436cb880f935f0bb48a6cbd6eee513c9)